wire-library/wire-gradle-plugin/src/main/kotlin/com/squareup/wire/gradle/WirePlugin.kt

```diff
@@ -45,104 +52,179 @@ class WirePlugin : Plugin<Project> {
           it.isCanBeConsumed = false
         }
 
-    project.tasks.register("generateProtos", WireTask::class.java) { task ->
-      task.group = "wire"
-      task.description = "Generate Wire protocol buffer implementation for .proto files"
-    }
-
-    project.plugins.all {
-      logger.debug("plugin: $it")
-      when (it) {
-        is JavaBasePlugin -> {
-          java = true
-        }
-        is KotlinBasePluginWrapper -> {
-          kotlin = true
-        }
+    val androidPluginHandler = { _: Plugin<*> ->
+      android.set(true)
+      project.afterEvaluate {
+        project.setupWireTasks(afterAndroid = true)
       }
     }
+    project.plugins.withId("com.android.application", androidPluginHandler)
+    project.plugins.withId("com.android.library", androidPluginHandler)
+    project.plugins.withId("com.android.instantapp", androidPluginHandler)
+    project.plugins.withId("com.android.feature", androidPluginHandler)
+    project.plugins.withId("com.android.dynamic-feature", androidPluginHandler)
```

Comment:
is this list of plugin IDs listed somewhere?

Reply:
I don't think so. I use what SqlDelight is using + the 2 javas. I looked around but didn't find anything.